rpcutils/rpcclient.go

```diff
@@ -448,6 +449,53 @@ func CommonAncestor(client *rpcclient.Client, hashA, hashB chainhash.Hash) (*cha
 	return &hashA, chainA, chainB, nil
 }
 
+// OrphanedTipLength finds a common ancestor by iterating block heights
+// backwards until a common block hash is found. Unlike CommonAncestor, an
+// orphaned DB tip whose corresponding block is not known to dcrd will not cause
+// cause an error. The number of blocks that have been orphaned is returned.
+// Realistically, this should rarely be anything but 0 or 1, but no limits are
+// placed here on the number of blocks checked.
+func OrphanedTipLength(ctx context.Context, client *rpcclient.Client,
+	tipHeight int64, hashFunc func(int64) (string, error)) (int64, error) {
+	commonHeight := tipHeight
+	var dbHash string
+	var err error
+	var dcrdHash *chainhash.Hash
+	for {
+		// Since there are no limits on the number of blocks scanned, allow
+		// cancellation for a clean exit.
+		select {
+		case <-ctx.Done():
+			return 0, nil
```

Comment:
If there is an interrupt, and this returns `0, nil` then I feel like there needs recognize this in main.  It could either be one of these:

```go
// Allow Ctrl-C to halt startup here.
	if shutdownRequested(ctx) {
return nil
```

or `OrphanedTipLength` could return -1 or a special non-nil error.